### DISCUSSION TITLE
Disk Size bei VM unter Proxmox

### DISCUSSION THREAD

**Discussion by ChristophCaina** [OpenCCU/OpenCCU#2002] [🇩🇪 Fragen&Antworten] [answered]
Hallo Zusammen,

ich habe gerade die Installation als VM unter Proxmox durchgeführt und dabei das Install-Script genutzt, welches in der Dokumentation genannt wird.

In meinem Fall wurde zwar die Version abgefragt, welche installiert werden sollte - ist dann aber, anders als in der Dokumentation geschildert - direkt zur Auswahl des USB Gerätes gesprungen.

Wonach nicht gefragt wurde, wo die Disk erstellt werden sollte (local-lvm) - dies wurde automatisch korrekt ausgewählt, aber wichtiger: Es wurde auch nicht nach der Größe der Disk gefragt.

Das Script hat die Disk mit 69G (64G Nutzbar) angelegt:

![grafik](https://user-images.githubusercontent.com/26391061/196341124-ce0b46cb-7318-45ac-b92e-d78ab2afa902.png)
![grafik](https://user-images.githubusercontent.com/26391061/196341132-1e061d5f-e3a4-4c39-9d6c-30029c38467a.png)

Das finde ich für ein System wie RaspberryMatic schon ein wenig arg viel ... und das verkleinern einer VM-Platte ist immer ein gewisser Akt unter Proxmox [vergrößern ist kein Problem, verkleinern leider nicht ganz so einfach]

**Comment by ChristophCaina** [comment]
Ich habe jetzt auch in der Dokumentation bei der alternativen Installation gesehen, dass hier eine Plattengröße von 64/65 G konfiguriert wurde...
Wenn ich das mit der VM vergleiche, welche standardmäßig von HomeAssistant erstellt wird (32G) dann erscheint mir das wirklich etwas groß... vielleicht bin ich aber dadurch verwöhnt, dass die meisten Container und Systeme mit deutlich kleineren Platten zurecht kommen (Standard sind für die reine Systemplatte zwischen 8 und 16 GB ausreichend...)

**Comment by Baxxy13** [comment, marked as answer]
Letztlich ist es ein virtuelles Dateisystem. Die 64GB sind das Maximum was die RM zur Verfügung hat. Physisch auf deinem Host dürfte die reale Größe so bei etwa 2GB liegen.

Du kannst natürlich die VM per Hand anlegen und die VHD Größe selbst bestimmen. Oder du speicherst dir das Script von Jens lokal ab, passt es deinen Vorstellungen an und führst es dann aus.

Meine RM-VM's haben alle nur 12GB VHD. 😉

**Comment by xxstasxx** [comment]
Könntest du mir das bitte ausführlicher erklären?
Danke.

**Reply by Baxxy13** [reply]
Was denn genau?
Wie man RM manuell unter Proxmox installiert (ohne Script-Hilfe) ist im Wiki beschrieben. ([Alternative/Manuelle Installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE#alternativemanuelle-installation))
Dort bei Punkt 6 steht wie man die Größe der VM-Disk ändert. Die initiale Disk hat ca. 1,35GB, um (etwa) auf meine 12GB zu kommen gibt man dort also 11GB ein.

Das mit derAnpassung des Install-Scriptes geht so:
- Script herunterladen: `wget https://raspberrymatic.de/install-proxmox.sh` 
- Script editieren (z.B. mit nano oder vi): `nano install-proxmox.sh`
- Hier https://github.com/jens-maus/RaspberryMatic/blob/e9af055c785697261f76badea7bd71b7adcb9b6c/scripts/install-proxmox.sh#L272 die 64G durch deine Wunschgröße ersetzen und speichern
- Script ausführen `bash ./install-proxmox.sh`

**Reply by xxstasxx** [reply]
Danke!
Genau das habe ich gebraucht, ich hatte vor den Script extern zu bearbeiten, aber so ist es ja viel einfacher.
